srcalc/src/main/java/gov/va/med/srcalc/domain/model/RiskModel.java

```diff
@@ -324,29 +325,73 @@ public double calculate(final Collection<Value> inputValues) throws MissingValue
         final List<MissingValueException> missingList = new ArrayList<MissingValueException>();
         for (final ModelTerm term : getTerms())
         {
-        	try
-        	{
-        		final double termSummand = term.getSummand(valueMap);
+            try
+            {
+                final double termSummand = term.getSummand(valueMap);
                 fLogger.debug("Adding {} for {}", termSummand, term);
                 sum += termSummand;
-        	}
+            }
             catch(final MissingValuesException e)
             {
-            	missingList.addAll(e.getMissingValues());
+                missingList.addAll(e.getMissingValues());
             }
         }
         
         if(missingList.size() > 0)
         {
-        	throw new MissingValuesException("The calculation is missing values.", missingList);
+            throw new MissingValuesException("The calculation is missing values.", missingList);
         }
         final double expSum = Math.exp(sum);
         
         return expSum / (1 + expSum);
     }
     
-    // TODO: implement equals() and hashCode()
-    
+    /**
+     * Returns a true if a RiskModel and if displayname and all Terms are equal.
+     */
+    @Override
+    public final boolean equals(final Object o)
+    {
+        if( this == o )
+        {
+            return true;
+        }
+        else if( o == null || !(o instanceof RiskModel) ) 
+        {
+            return false;
+        }
+        else 
+        {
+            // don't include the Id (Hibernate best practices)
+            final RiskModel other = (RiskModel)o;
+            
+            if( this.getDisplayName().compareTo( other.getDisplayName() ) != 0 ) 
+            {
+                return false;
+            }        
+            
+            return Objects.equals( getTerms(), other.getTerms() );
+        }
+    }
+
+    /**
+     * Returns a hash code based on the displayName that uses the String.hashCode() implementation.
```

Comment:
If you read the comment, it says "Returns a hash code based on the displayName that uses the String.hashCode() implementation."